smack-experimental/src/main/java/org/jivesoftware/smackx/httpfileupload/exception/HttpUploadException.java

```diff
@@ -0,0 +1,42 @@
+/**
+ *
+ * Copyright the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jivesoftware.smackx.httpfileupload.exception;
+
+import java.io.IOException;
+import java.net.URL;
+
+import org.jivesoftware.smackx.httpfileupload.element.Slot;
+
+/**
+ * An exception class to provide additional information in case of exceptions during file uploading.
+ *
+ */
+public class HttpUploadException extends IOException {
```

Comment:
Creating a new Java package for a single class seems like overkill. Just place it in o.j.smackx.httpfileupload. Also, what's the point if the constructors have parameters, but the arguments of the parameters are never accessible by the user? I think this class is missing some `private final` fields and the according getters.

Reply:
Fair enough, I made the class more accessible with appropriate getters